xtra-libp2p-offer/src/taker.rs

```diff
@@ -0,0 +1,71 @@
+use crate::protocol;
+use async_trait::async_trait;
+use model::MakerOffers;
+use xtra::prelude::MessageChannel;
+use xtra::spawn::TokioGlobalSpawnExt;
+use xtra::Actor as _;
+use xtra_libp2p::NewInboundSubstream;
+use xtra_productivity::xtra_productivity;
+use xtras::spawner;
+use xtras::spawner::SpawnFallible;
+use xtras::SendAsyncSafe;
+
+pub struct Actor {
+    maker_offers: Box<dyn MessageChannel<LatestMakerOffers>>,
+    spawner: xtra::Address<spawner::Actor>,
+}
+
+impl Actor {
+    pub fn new(maker_offers: &(impl MessageChannel<LatestMakerOffers> + 'static)) -> Self {
+        let spawner = spawner::Actor::new().create(None).spawn_global();
+
+        Self {
+            maker_offers: maker_offers.clone_channel(),
+            spawner,
+        }
+    }
+}
+
+#[xtra_productivity(message_impl = false)]
+impl Actor {
```

Comment:
No need to action this - some ideas:

Somehow in my head I had turned the pattern around and would have let the taker open a substream and the maker sending the current offers upon incoming substream. i.e. poll, not push.

This is different to what we have now, but it has the advantage that we could more easily implement a feature where inactive takers "go to sleep" and the maker does not have to bother sending offers to them.
When it comes to resilience and stability I feel it might be more resilient, because each taker tries to get an offer and reconnects and the maker does not have to care about a peer being unavailable at this point. 

Not sure if it's worth thinking about this now though. I'm happy to roll this out using this pattern and potentially change it later.

Reply:
I followed the design in the proposal as much as possible. It states:

> The maker knows when the offer changes, pushing the latest offer is more efficient.

You bring up some interesting points, but I would say introducing a new design for this protocol is out of the scope of this PR.